erpnext/public/js/controllers/taxes_and_totals.js

```diff
@@ -754,5 +754,51 @@ erpnext.taxes_and_totals = erpnext.payments.extend({
 				}
 			});
 		})
+	},
+	set_and_update_cultivation_tax: function () {
+		const me = this;
+		if (["Quotation", "Sales Invoice", "Delivery Note", "Sales Order"].includes(me.frm.doctype) && !me.frm.doc.license) {
+			return;
+		}
+
+		frappe.db.get_value("Compliance Info", { "name": me.frm.doc.license }, "license_for", (r) => {
+			if (["Sales Order", "Sales Invoice", "Delivery Note", "Stock Entry"].includes(me.frm.doctype) && (!r || r.license_for !== "Distributor")) {
+				return;
+			}
+
+			frappe.call({
+				method: "erpnext.compliance.taxes.set_cultivation_tax",
+				args: {
+					doc: me.frm.doc
+				},
+				callback: (r) => {
+					if (r.message) {
+						let taxes = me.frm.doc.taxes;
+						// get tax account head from company master to avoid disorder behaviour and remove it from taxes table.
+						frappe.db.get_value('Company', { "name": me.frm.doc.company }, ["default_cultivation_tax_account_flower", "default_cultivation_tax_account_leaf", "default_cultivation_tax_account_plant"], (r) => {
+							let tax_account_list = [r.default_cultivation_tax_account_flower, r.default_cultivation_tax_account_leaf, r.default_cultivation_tax_account_plant];
+							if (taxes && taxes.length > 0) {
+								$.each(taxes, function (i, tax) {
+									if (tax_account_list.indexOf(tax.account_head) !== -1) {
+										me.frm.get_field("taxes").grid.grid_rows[i].remove();
+									}
+								});
+							}
+						});
```

Comment:
- Rather than multiple `frappe.call`, just use one frappe call and do the manipulations in the backend.
- frappe.db.get_value internally does a frappe.call too